### PR TITLE
Auth0 Integration V2

### PR DESCRIPTION
**Pull request by aidenprice**
Hi again,

This is the cleaned up PR from the previous attempt, please ignore my earlier PR and consider this one instead. Sorry for the messy git history.

This PR adds Auth0 OAuth integration following the same pattern as Okta. The key names are different enough to justify a different handler.

Thank you again,
Aiden

### CODE REVIEW

**Comment by jczhong84** [comment]
@aidenprice thanks for the PR to add Auth0. 
Looks like it's mostly same as okta except the `get_auth0_urls`. Maybe we can just inherit it from `OktaLoginManager`,  rename `get_okta_urls` as `get_oauth_urls` and override it for `Auth0LoginManager`

**Comment by aidenprice** [comment]
Hi, I've tried your suggestion and it certainly cleans up a lot of repeated code.
One thing I should point out though, the `LOG = get_logger(__file__)` isn't inheritable or injectable. So when you instantiate an `Auth0LoginManager` it will represent logs as coming from the Okta file.

File: querybook/server/app/auth/auth0_auth.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import requests
+from env import get_env_config
+
+from app.auth.okta_auth import OAUTH_CALLBACK_PATH, OktaLoginManager
+
+
+class NoopAuth(requests.auth.AuthBase):
```
Comment (jczhong84) [inline querybook/server/app/auth/auth0_auth.py:7]: this is not used

Comment (aidenprice) [reply querybook/server/app/auth/auth0_auth.py]: Done, removed it.

**Review by jczhong84 — APPROVED** [review]

**Comment by aidenprice** [comment]
Thank you!

**Pull request merged**